#!/usr/bin/env python
PACKAGE = "franka_effort_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 600, 100, 10000)
# gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 1600, 0, 10000)
# gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 1600, 0, 10000)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 1200, 16, 10000)
gen.add("cartesian_damping_ratio", double_t, 0, "Cartesian damping ratio", 0.7, 0.5, 5)
gen.add("mass", double_t, 0, "Cartesian mass matrix", 3, 1, 100)
# gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller", 16, 0, 100)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller", 3, 0, 100)
gen.add("target_k_p", double_t, 0, "Force_control_PI_controller_kp", 0.5, 0, 100)
gen.add("target_k_i", double_t, 0, "Force_control_PI_controller_ki", 1.2, 0, 100)

exit(gen.generate(PACKAGE, "dynamic_compliance", "CarImpParam"))
